modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -336,7 +337,8 @@ protected void finishHim(Exception failure, List<Failure> indexingFailures, List
             final XContentType mainRequestXContentType = mainRequest.getDestination().getContentType();
             if (mainRequestXContentType != null && doc.getXContentType() != mainRequestXContentType) {
                 // we need to convert
-                try (XContentParser parser = sourceXContentType.xContent().createParser(NamedXContentRegistry.EMPTY, doc.getSource());
+                try (XContentParser parser = sourceXContentType.xContent()
+                        .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, doc.getSource());
```

Comment:
I believe we can make this the throwing one because we just copy the structure.

Reply:
Yeah think so, since there isn't such a think as a deprecated field in the body of a document